src/Castle.Core/DynamicProxy/DefaultProxyBuilder.cs

```diff
@@ -104,11 +109,24 @@ public Type CreateClassProxyType(Type classToProxy, Type[] additionalInterfacesT
 		{
 			AssertValidType(interfaceToProxy);
 			AssertValidTypes(additionalInterfacesToProxy);
+			AssertValidMixins(options, nameof(options));
 
 			var generator = new InterfaceProxyWithoutTargetGenerator(scope, interfaceToProxy) { Logger = logger };
 			return generator.GenerateCode(typeof(object), additionalInterfacesToProxy, options);
 		}
 
+		private void AssertValidMixins(ProxyGenerationOptions options, string paramName)
+		{
+			try
+			{
+				options.Initialize();
```

Comment:
(Up until now, `options` was initialized inside the various generator classes' `GenerateCode` methods. If we're now initializing one level further up in the call hierarchy, those calls (albeit cheap) will become redundant. I'm planning to clean up redundant calls to `options.Initialize` in a future PR for #517.)